src/common/reducers/repository-input.js

```diff
@@ -19,39 +13,15 @@ export function repositoryInput(state = {
       return {
         ...state,
         inputValue: action.payload,
-        repository: parseRepository(action.payload),
+        repository: parseGitHubRepoUrl(action.payload),
```

Comment:
I think we should **both** create a reducer for repository information and retain repository information inside the repository input form reducer.

A repository info reducer would be good for pages where the repository is an important context - the build details page, for example.

The repository information in the repository input form reducer is responsible for storing the state of the form field. I think the distinction is important, because the form data can be empty, invalid or incomplete. The data describes something which _might be_ a repository, whereas a repository info reducer would represent a _concrete_ repository.

This was a common occurrence on the last app I worked on. We had a "store" (a reducer in Redux terms) for payment methods when they were important for the page being viewed (e.g. making a payment with a saved credit card). We had a separate "store" for the new payment method form. Both used a standardised shape for the payment methods, and we extracted common methods for validating, transforming and decorating credit card data into shared helper functions.